### ISSUE TITLE
Ribbon ConnectTimeout and ReadTimeout setting bug

### ISSUE THREAD

**Issue by esolve** [spring-cloud/spring-cloud-netflix#3786]
**Describe the bug**
in class "org.springframework.cloud.netflix.ribbon.RibbonClientConfiguration", we have the following code snippet which set Ribbon ConnectTimeout and ReadTimeout

```
    @Bean
    @ConditionalOnMissingBean
    public IClientConfig ribbonClientConfig() {
        DefaultClientConfigImpl config = new DefaultClientConfigImpl();
        config.loadProperties(this.name);
        config.set(CommonClientConfigKey.ConnectTimeout, 1000);
        config.set(CommonClientConfigKey.ReadTimeout, 1000);
        return config;
    }
```
it loads ConnectTimeout and ReadTimeout setting  from application.yml/application.properties, but then override ConnectTimeout and ReadTimeout with value of 1000, this makes these two values in application.yml/application.properties useless

**Comment by 970655147** [comment]
there is `DynamicProperty` in DefaultClientConfigImpl, it might be helpful

**Comment by FelipeAdorno** [comment]
Can I help with this issue? Which branch I can use to fix?

**Comment by spencergibb** [comment]
@FelipeAdorno sure, it would go in the 2.2.x branch.

**Comment by FelipeAdorno** [comment]
Ok, I'll work on it!

**Comment by spencergibb** [comment]
Closed via #3965
